boss-ws/src/main/java/org/genomebridge/boss/http/service/DatabaseBossAPI.java

```diff
@@ -296,23 +318,26 @@ public ErrorDesc getResumableUploadURL(String objectId, String userName, CopyRes
             return  writePermsErr(objectId,userName);
         }
         ObjectStore objStore = getObjectStore(rec.storagePlatform);
+        if (objStore != null && objStore.isReadOnly())
+            return readOnlyStoreErr("readOnlyStore");
+
         resp.uri = objStore.generateResumableUploadURL(rec.objectName);
         return error;
-
     }
-
     private BossDAO getDao() {
         return mDBI.onDemand(BossDAO.class);
     }
 
     private ObjectStore getObjectStore( String storagePlatform ) {
```

Comment:
This method is no longer useful.  It would be better to inline it in all the places it's called.  Then it will be easy to return an ErrorDesc for an invalid storagePlatform.